dvc/remote/local/__init__.py

```diff
@@ -573,3 +575,54 @@ def _get_unpacked_dir_names(self, checksums):
             if self.is_dir_checksum(c):
                 unpacked.add(c + self.UNPACKED_DIR_SUFFIX)
         return unpacked
+
+    def _needs_checkout(self, path_info, checksum, force):
+        # NOTE: In case if path_info is already cached, and cache type is
+        # 'copy' we would like to avoid relinking.
+        # `symlink/hardlink/reflink` is cheap, but if we need `copy` links,
+        # then to save on copying we should only relink if we already have
+        # `hardlink` or a `symlink`, we don't care about reflinks, because
+        # they are indistinguishable from copy anyway.
+
+        if (
+            not force
+            and not self.changed(path_info, {self.PARAM_CHECKSUM: checksum})
+            and not System.is_hardlink(path_info)
+            and not System.is_symlink(path_info)
+            and self._is_cache_copy(path_info)
+        ):
+            if self.protected:
+                self.protect(path_info)
+            else:
+                self.unprotect(path_info)
```

Comment:
Oh, btw, this one will create copies, right? We've discussed that it shouldn't for this purpose.

Reply:
It should not, because it is execute after `is_symlink` and after `is_hardlink`. So in this situation it should only perform `chmod`.

Reply:
Ah, right! Good point! Let's also leave a comment about it.

Reply:
done